File: Java/practice2/src/DataSources/Data/DatabaseConnection.java
package DataSources.Data;

import io.github.cdimascio.dotenv.Dotenv;
import io.github.cdimascio.dotenv.DotenvBuilder;
import io.github.cdimascio.dotenv.DotenvEntry;

import java.sql.*;
import java.util.Set;
import java.util.stream.Stream;

public class DatabaseConnection {

    public static boolean connectionStatus;
    public static Connection CONNECTION;


    public static Boolean connect(boolean testEnvironment) {
        Dotenv vault = Dotenv.load();
        String user = vault.get("MARIADB_USER");
        String url = vault.get("MARIADB_DATABASE_URL");
        String database = vault.get("SET_DATABASE");
        String testDatabase = vault.get("SET_TEST_DATABASE");
        //Connection dbConnection;

        try{
            Class<?> driver = Class.forName(vault.get("MARIADB_JDBC_DRIVER"));
            CONNECTION = DriverManager.getConnection(url, user,"");
            PreparedStatement setDatabase;
            String dbSelected;
            if(testEnvironment) {
                setDatabase = CONNECTION.prepareStatement("USE " + testDatabase + ";");
                dbSelected = testDatabase;
            } else{
                setDatabase = CONNECTION.prepareStatement("USE " + database + ";");
                dbSelected = database;
            }
            setDatabase.execute();
            connectionStatus = true;
            System.out.println("Using: " + database + " | Connection status: " + connectionStatus);
        }catch (SQLException e){
            System.out.println(e.getMessage());
            e.printStackTrace();
            System.out.println("Connection failed OR closed connection.");
        } catch (ClassNotFoundException e){
            System.out.println(e.getMessage());
            e.printStackTrace();
            System.out.println("Class.forName failed to recognize driver class.");
        }

        return connectionStatus;

    }
}

File: Kotlin/practice2/src/DataSources/Data/DatabaseConnection.kt
package DataSources.Data

import io.github.cdimascio.dotenv.Dotenv
import io.github.cdimascio.dotenv.dotenv
import java.sql.Connection
import java.sql.DriverManager
import java.sql.PreparedStatement
import java.sql.SQLException



class DatabaseConnection {

    companion object{
        var connectionStatus: Boolean = false
        val CONNECTION: Connection = this.connect("Practice2")


        fun connect(database: String) : Connection{
            val vault: Dotenv = dotenv()
            val user: String? = vault["MARIADB_USER"]
            val url: String? = vault["MARIADB_DATABASE_URL"]

            lateinit var dbConnection: Connection

            try{
                val driver = Class.forName(vault["MARIADB_JDBC_DRIVER"])
                dbConnection = DriverManager.getConnection(url, user, "")
            }catch (e: SQLException){
                e.message
                e.printStackTrace()
                println("Connection failed")
            }catch (e: ClassNotFoundException){
                e.message
                e.printStackTrace()
                println("Class.forName failed to recognize driver class")
            }

            val setDatabase: PreparedStatement = dbConnection.prepareStatement("USE $database;")

            try{
                val checkOpen = dbConnection.isClosed()
                setDatabase.execute()
                this.connectionStatus = true
                println("Using: $database | Connection Status: ${connectionStatus}")
            }catch (e: SQLException) {
                e.message
                e.printStackTrace()
                println("Database is closed and must be opened before query executions OR database don't exist.")
            }

            return dbConnection

        }

    }
    fun checkStatus() : Boolean {
        return connectionStatus
    }
}